### PR TITLE
FileLineCountSniff

### LINKED ISSUE #1290: request:  FileLengthSniff
LIke FunctionLengthSniff, but for the entire file

CodeClimate's default threshold is 250 lines (not including comments / whitespace)
https://docs.codeclimate.com/docs/default-analysis-configuration

> File lines
Threshold: 250

### PR DESCRIPTION
**Pull request by bkdotcom**
Adds new sniff:
* SlevomatCodingStandard.Files.**FileLength**

has same options as existing SlevomatCodingStandard.Functions.FunctionLength
* maxLinesLength = 250
* includeComments = false
* includeWhitespace = false

resolves #1290

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -773,6 +773,14 @@ Sniff provides the following settings:
 
 However, if you prefer Yoda conditions, you can use `RequireYodaComparisonSniff`.
 
+#### SlevomatCodingStandard.Files.FileLineCount
```
Comment (kukulich) [inline README.md:776]: Is there a reason why not to be consistent and name it `FileLength` ?

File: SlevomatCodingStandard/Sniffs/Files/FileLineCountSniff.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+			FunctionHelper::LINE_INCLUDE_COMMENT => $this->includeComments,
+			FunctionHelper::LINE_INCLUDE_WHITESPACE => $this->includeWhitespace,
+		]));
+		$flags = array_reduce($flags, static function ($carry, $flag): int {
+			return $carry | $flag;
+		}, 0);
+
+		$length = FunctionHelper::getLineCount($phpcsFile, $pointer, $flags);
+
+		if ($length <= SniffSettingsHelper::normalizeInteger($this->maxLinesLength)) {
```
Comment (kukulich) [inline SlevomatCodingStandard/Sniffs/Files/FileLineCountSniff.php:53]: Please normalize the setting on the start of the `process` method, see https://github.com/slevomat/coding-standard/commit/0e4df34a76d26d3964c7654515b97a5b0f4cfb84

**Review by kukulich — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -949,7 +957,7 @@ Disallows numeric literal separators.
 
 Requires use of numeric literal separators.
 
-This sniff provides the following setting:
+This sniff provides the following settings:
```
Comment (kukulich) [inline README.md:960]: These fixes are great but should be in a separate commit.

File: README.md (hunk 1 above)
Comment (bkdotcom) [reply README.md:776]: My reasoning was to further "disambiguate" from Files/LineLengthSniff 

LineLengthSniff   // sniffs # of columns in line  (not number of lines)
FileLengthSniff   // sniffs # of lines in file    ( feel like  "LineCountSniff"  is more descriptive  ?)

Comment (kukulich) [reply README.md:776]: However have FunctionLengthSniff and it checks number of liness.

Comment (bkdotcom) [reply README.md:776]: 👍 / will rename

Comment (bkdotcom) [reply README.md]: done

**Comment by bkdotcom** [comment]
@kukulich
once again thanks
All requested changes should be resolved

**Comment by kukulich** [comment]
Thanks.
